intl_translation_format/README.md

```diff
@@ -56,6 +56,169 @@ pub run intl_translation_format:generate
+ 4. How does it handle Plurals, Genders, and Variables?
+
+Let's answer these questions for our new JSON format:
+ 1. While it is a standard data interchange format, there is no official standard for translations. Multiple frameworks create their implementations like `i18next` or `ng2-translate` for `Angular`.
+
+ Because the `ICU Format` is the core of the dart intl library, we decided to create a value/key simple JSON with ICU messages. 
+
+ As an example: 
+ ```json
+ {
+   "message1" : "This is a message",
+   "message2" : "Hello { userName }, this a message with a variable",
+   "message2" : "You have { howMany, plural, =0 { none Apples} one { one Apple} other { {howMany} Apples} }"
+ } 
+ ```
+ 2. It is a text-based format
+ 3. It contains one locale translation per file.
+ 4. We will use the ICU standard for plurals, genders, and variables.
+
+
+2. Start coding
+
+ - Create a new dart package and add intl_translation_format as a dependency.
+
+- Choose what `TranslationFormat` class to extend. In our case, our format will be monolingual and text-based so we will extend `MonolingualFormat`. You can see the implementation of other classes in other packages.
+
+```dart 
+class JsonFormat extends MonoLingualFormat {
+
+  @override
+  String get fileExtension => 'json';
+  ...
+}  
+```
+The first thing will be to add the `fileExtension` in our case `JSON`. You can override the `fileExtensions` list param in case your format supports multiple extensions (eg. Xliff supports .xliff and xlf).
+
+3. Implement the method to generate a template.
+
+This method will generate a file in the desired format that will contain the messages from the template.
+
+```dart
+@override
+  String generateTemplateFile(
+    TranslationTemplate catalog,
+  ) {
+    final messages = catalog.messages;
+    var json = '{\n';
+    messages.forEach((key, value) {
+      // Expands the MainMessage into an icu string
+      final message = messageToIcuString(value);
+      json += '  "$key": "$message",\n';
+    });
+    if (messages.isNotEmpty) json = json.substring(0, json.length - 2) + '\n';
+    json += '}';
+    return json;
+  }
+```
+The param `catalog` contains the list of `MainMessage`s. These `MainMessages` can be expanded into a string with your transform function. For ICU messages there is an already built method `messageToIcuString` that expands the message into an ICU string.
+
+This method should return a String that contains the file content with the desired format.
+
+4. Implement the method to extract messages: 
+
+It will return `MessagesForLocale` object with the messages and an optional locale in case it is indicated inside the file.
+
+```dart
+@override
+  MessagesForLocale parseFile(
+    String content, {
+    MessageGeneration generation,
+  }) {
+    final json = jsonEncoder.decode(content) as Map<String, Object>;
+    final messages = <String, BasicTranslatedMessage>{};
+     // foldMap Creates a flat map with a set of keys per value.
```

Comment:
I don't know what this means? Maybe pull the foldMap part into a separate variable? And/or put an example in the comment of what it looks like or what it means? Set of keys per value sounds like
  {
    { 'a', 'b', 'c'} : 'the value'
}

What are the keys and values in this case? Is that the parameter names and the ICU message? Is it misleading for this to be a Map, since it doesn't seem like we ever do key lookup?